### ISSUE TITLE
Can't start grafana-server daemon.

### ISSUE THREAD

**Issue by Nihn** [grafana/grafana#1835]
Hi,

i can't start grafana-server daemon, output looks like this ('failed' start coming after ~10s and doesnt stop until ^C) :
```
root:/home/vagrant# service grafana-server start
[FAIL] Starting Grafana Server: failed!
 failed!
 failed!
 failed!
 failed!
 failed!
 failed!
```
After first attempt grafana.log contains:
```
015/04/21 08:09:58 [I] Starting Grafana
2015/04/21 08:09:58 [I] Version: 2.0.1, Commit: 9540e12, Build date: 2015-04-20 04:21:42 -0700 PDT
2015/04/21 08:09:58 [I] Configuration Info
Config files:
  [0]: /usr/share/grafana/conf/defaults.ini
  [1]: /etc/grafana/grafana.ini
Command lines overrides:
  [0]: default.paths.data=/var/lib/grafana
  [1]: default.paths.logs=/var/log/grafana
Paths:
  home: /usr/share/grafana
  data: /var/lib/grafana
  logs: /var/log/grafana

2015/04/21 08:09:58 [I] Database: sqlite3, ConnectionString: file:/var/lib/grafana/grafana.db?cache=shared&mode=rwc&_loc=Local
2015/04/21 08:09:58 [I] Migrator: Starting DB migration
2015/04/21 08:09:58 [I] Migrator: exec migration id: create migration_log table
2015/04/21 08:09:58 [I] Migrator: exec migration id: create user table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index user.login
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index user.email
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_user_login - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_user_email - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: Rename table user to user_v1 - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create user table v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_user_login - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_user_email - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy data_source v1 to v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: Drop old table user_v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create star table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index star.user_id_dashboard_id
2015/04/21 08:09:58 [I] Migrator: exec migration id: create org table v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_org_name - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create org_user table v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index IDX_org_user_org_id - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_org_user_org_id_user_id - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy data account to org
2015/04/21 08:09:58 [I] Migrator: skipping migration id: copy data account to org, condition not fulfilled
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy data account_user to org_user
2015/04/21 08:09:58 [I] Migrator: skipping migration id: copy data account_user to org_user, condition not fulfilled
2015/04/21 08:09:58 [I] Migrator: exec migration id: Drop old table account
2015/04/21 08:09:58 [I] Migrator: exec migration id: Drop old table account_user
2015/04/21 08:09:58 [I] Migrator: exec migration id: create dashboard table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add index dashboard.account_id
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index dashboard_account_id_slug
2015/04/21 08:09:58 [I] Migrator: exec migration id: create dashboard_tag table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index dashboard_tag.dasboard_id_term
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_dashboard_tag_dashboard_id_term - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: Rename table dashboard to dashboard_v1 - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create dashboard v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index IDX_dashboard_org_id - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_dashboard_org_id_slug - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy dashboard v1 to v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop table dashboard_v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create data_source table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add index data_source.account_id
2015/04/21 08:09:58 [I] Migrator: exec migration id: add unique index data_source.account_id_name
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index IDX_data_source_account_id - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_data_source_account_id_name - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: Rename table data_source to data_source_v1 - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create data_source table v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index IDX_data_source_org_id - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_data_source_org_id_name - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy data_source v1 to v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: Drop old table data_source_v1 #2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create api_key table
2015/04/21 08:09:58 [I] Migrator: exec migration id: add index api_key.account_id
2015/04/21 08:09:58 [I] Migrator: exec migration id: add index api_key.key
2015/04/21 08:09:58 [I] Migrator: exec migration id: add index api_key.account_id_name
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index IDX_api_key_account_id - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_api_key_key - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop index UQE_api_key_account_id_name - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: Rename table api_key to api_key_v1 - v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create api_key table v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index IDX_api_key_org_id - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_api_key_key - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: create index UQE_api_key_org_id_name - v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: copy api_key v1 to v2
2015/04/21 08:09:58 [I] Migrator: exec migration id: Drop old table api_key_v1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create dashboard_snapshot table v4
2015/04/21 08:09:58 [I] Migrator: exec migration id: drop table dashboard_snapshot_v4 #1
2015/04/21 08:09:58 [I] Migrator: exec migration id: create dashboard_snapshot table v5 #2
2015/04/21 08:09:59 [I] Migrator: exec migration id: create index UQE_dashboard_snapshot_key - v5
2015/04/21 08:09:59 [I] Migrator: exec migration id: create index UQE_dashboard_snapshot_delete_key - v5
2015/04/21 08:09:59 [I] Migrator: exec migration id: create index IDX_dashboard_snapshot_user_id - v5
2015/04/21 08:09:59 [I] Created default admin user: admin
```

After next attempts it's only contains: (bash output remains the same)
```
2015/04/21 08:10:50 [I] Starting Grafana
2015/04/21 08:10:50 [I] Version: 2.0.1, Commit: 9540e12, Build date: 2015-04-20 04:21:42 -0700 PDT
2015/04/21 08:10:50 [I] Configuration Info
Config files:
  [0]: /usr/share/grafana/conf/defaults.ini
  [1]: /etc/grafana/grafana.ini
Command lines overrides:
  [0]: default.paths.data=/var/lib/grafana
  [1]: default.paths.logs=/var/log/grafana
Paths:
  home: /usr/share/grafana
  data: /var/lib/grafana
  logs: /var/log/grafana

2015/04/21 08:10:50 [I] Database: sqlite3, ConnectionString: file:/var/lib/grafana/grafana.db?cache=shared&mode=rwc&_loc=Local
2015/04/21 08:10:50 [I] Migrator: Starting DB migration
```
I tested it with sqllite3 and postgres, with same result.
I have Grafana 2.0.1 on Debain Wheezy 64bit.

**Comment by torkelo** [comment]
looks really strange, like like you have not gotten any config files loaded. This works on all Wheezy images I have tested with. How did you install the package, have you done any changes to the init.d script or config file?

**Comment by torkelo** [comment]
Ignore the part where I said it looks like no config files are loaded, was just github that did not show your lops right the first time I viewed this issue. 

Did you make any config changes to /etc/grafana/grafana.ini ?

**Comment by Nihn** [comment]
@torkelo yes, i changed port to 80 and session provider to Redis. Rest are defaults.

**Comment by Nihn** [comment]
I've checked with defaults and it seems to grafana-server can't run with port 80.

**Comment by torkelo** [comment]
@Nihn sure it can, its just that to run on port 80 Linux requires some extra user privalages  (usually need to be root or do some other security changes). Its in the docs: http://docs.grafana.org/installation/configuration/#http_port

**Comment by Nihn** [comment]
@torkelo  thanks for an answer, i didn't see that :). But anyway I think that behaviour with mutliple 'fail' output of init script isn't desired. When service is starting e.g. with puppet then it causes whole provision to freeze.

**Comment by torkelo** [comment]
@Nihn the multiple fail output is fixed in v2.0.2
